File: DeviceParts/src/com/aicp/device/DeviceSettings.java
/*
* Copyright (C) 2016 The OmniROM Project
*
* This program is free software: you can redistribute it and/or modify
* it under the terms of the GNU General Public License as published by
* the Free Software Foundation, either version 2 of the License, or
* (at your option) any later version.
*
* This program is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
* GNU General Public License for more details.
*
* You should have received a copy of the GNU General Public License
* along with this program. If not, see <http://www.gnu.org/licenses/>.
*
*/
package com.aicp.device;

import android.os.Bundle;
import androidx.preference.PreferenceFragment;
import androidx.preference.Preference;
import androidx.preference.TwoStatePreference;

public class DeviceSettings extends PreferenceFragment implements
        Preference.OnPreferenceChangeListener {

    public static final String KEY_VIBSTRENGTH = "vib_strength";
    public static final String KEY_FASTCHARGE = "fastcharge";
    public static final String KEY_SETTINGS_PREFIX = "device_setting_";

    private VibratorStrengthPreference mVibratorStrength;
    private static TwoStatePreference mFastChargeSwitch;

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.main, rootKey);

        mVibratorStrength = (VibratorStrengthPreference) findPreference(KEY_VIBSTRENGTH);
        if (mVibratorStrength != null) {
            mVibratorStrength.setEnabled(VibratorStrengthPreference.isSupported());
        }
        mFastChargeSwitch = (TwoStatePreference) findPreference(KEY_FASTCHARGE);
        mFastChargeSwitch.setEnabled(FastChargeSwitch.isSupported());
        mFastChargeSwitch.setChecked(FastChargeSwitch.isCurrentlyEnabled(this.getContext()));
        mFastChargeSwitch.setOnPreferenceChangeListener(new FastChargeSwitch(getContext()));
    }

    @Override
    public boolean onPreferenceTreeClick(Preference preference) {
        return super.onPreferenceTreeClick(preference);
    }
    @Override
    public boolean onPreferenceChange(Preference preference, Object newValue) {
        return true;
    }
}

File: DeviceParts/src/com/aicp/device/DeviceSettings.kt
/*
* Copyright (C) 2016 The OmniROM Project
*
* This program is free software: you can redistribute it and/or modify
* it under the terms of the GNU General Public License as published by
* the Free Software Foundation, either version 2 of the License, or
* (at your option) any later version.
*
* This program is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
* GNU General Public License for more details.
*
* You should have received a copy of the GNU General Public License
* along with this program. If not, see <http://www.gnu.org/licenses/>.
*
*/
package com.aicp.device

import android.os.Bundle
import androidx.preference.Preference
import androidx.preference.PreferenceFragment
import androidx.preference.TwoStatePreference

class DeviceSettings : PreferenceFragment(), Preference.OnPreferenceChangeListener {
    private var mVibratorStrength: VibratorStrengthPreference? = null
    fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.main, rootKey)
        mVibratorStrength = findPreference(KEY_VIBSTRENGTH) as VibratorStrengthPreference?
        if (mVibratorStrength != null) {
            mVibratorStrength.setEnabled(VibratorStrengthPreference.isSupported())
        }
        mFastChargeSwitch = findPreference(KEY_FASTCHARGE) as TwoStatePreference?
        mFastChargeSwitch.setEnabled(FastChargeSwitch.isSupported())
        mFastChargeSwitch.setChecked(FastChargeSwitch.isCurrentlyEnabled(this.getContext()))
        mFastChargeSwitch.setOnPreferenceChangeListener(FastChargeSwitch(getContext()))
    }

    fun onPreferenceTreeClick(preference: Preference?): Boolean {
        return super.onPreferenceTreeClick(preference)
    }

    fun onPreferenceChange(preference: Preference?, newValue: Any?): Boolean {
        return true
    }

    companion object {
        const val KEY_VIBSTRENGTH = "vib_strength"
        const val KEY_FASTCHARGE = "fastcharge"
        const val KEY_SETTINGS_PREFIX = "device_setting_"
        private var mFastChargeSwitch: TwoStatePreference? = null
    }
}